File: ass1/ass1.cs
using System;
using System.Collections.Generic;
public class Ass1 {
	static List<string> tree(int n) {
		if (n == 1) {
			List<string> s = new List<string>();
			s.Add("(.)");
			return s;
		}

		List<string> strings = new List<string>();
		List<string> prev = tree(n - 1);
		foreach (string T in prev) {
			strings.Add("(" + T + ".)");
	        strings.Add("(." + T + ")");
		}
		return strings;
	}

    static void Main() {
        Console.Write("Enter the number of nodes: ");
        List<string> treeList = tree(Convert.ToInt32(Console.ReadLine()));
        foreach (string tree in treeList) {
            Console.WriteLine(tree);
        }
    }
}

// mcs -out:hello.exe hello.cs

File: ass1/ass1.hs
-- Running instructions 
-- gchi ass1.hs
-- main

tree :: Integer -> [String]

tree n = 
    if n == 1
        then ["(.)"]
        else 
            map (\t -> "(" ++ t ++ ".)") (tree (n-1)) ++ map (\t -> "(." ++ t ++ ")") (tree (n-1))


main = do
    putStrLn "Enter the number of nodes: "
    n <- getLine
    let num = (read n :: Integer)
    mapM_ putStrLn (tree num)

